test/e2e/e2e-upgrade.test.ts

```diff
@@ -0,0 +1,167 @@
+  let cheVersion: string
+
+  describe('Prepare latest stable Che', () => {
+    it(`Deploy Che using ${INSTALLER} installer and self signed certificates`, async () => {
+      // Retrieve latest stable Che version
+      const githubClient = new CheGithubClient()
+      const latestStableCheTag = (await githubClient.getTemplatesTagInfo(INSTALLER, 'latest'))!
+      cheVersion = latestStableCheTag.name
+
+      const deployCommand = `${binChectl} server:deploy --platform=${PLATFORM} --installer=${INSTALLER} --version=${cheVersion} --chenamespace=${NAMESPACE} --telemetry=off --che-operator-cr-patch-yaml=test/e2e/resources/cr-patch.yaml`
+      await runCommand(deployCommand)
+
+      expect(await waitForVersionInCheCR(cheVersion, CHE_VERSION_TIMEOUT_MS)).equal(true)
+    })
+
+    it('Prepare test workspace', async () => {
+      await loginTest()
+
+      // Create
+      await runCommand(binChectl, ['workspace:create', `--devfile=${DEVFILE_URL}`, '--telemetry=off', `-n ${NAMESPACE}`])
+      const workspaceId = await helper.getWorkspaceId()
+
+      // Start
+      await runCommand(binChectl, ['workspace:start', workspaceId, `-n ${NAMESPACE}`, '--telemetry=off'])
+      expect(await helper.waitWorkspaceStatus('RUNNING', WORKSPACE_START_TIMEOUT_MS)).to.equal(true)
+
+      // Stop
+      await runCommand(binChectl, ['workspace:stop', workspaceId, `-n ${NAMESPACE}`, '--telemetry=off'])
+      const workspaceStatus = await helper.getWorkspaceStatus()
+      // The status could be STOPPING or STOPPED
+      expect(workspaceStatus).to.contain('STOP')
+    })
+  })
+
+  describe('Test Che update', () => {
+    it('Update Che to nightly version', async () => {
+      await runCommand(binChectl, ['server:update', '-y', `-n ${NAMESPACE}`, '--telemetry=off'])
+      expect(await waitForCheServerImageTag(NIGHTLY, UPDATE_CHE_TIMEOUT_MS)).equal(true)
+    })
+
+    it('Check updated Che version', async () => {
+      expect(await waitForVersionInCheCR(NIGHTLY, CHE_VERSION_TIMEOUT_MS)).equal(true)
+    })
+  })
+
+  describe('Test updated Che', () => {
+    it('Start existing workspace after update', async () => {
+      // Relogin
+      await loginTest()
+
+      const workspaceId = await helper.getWorkspaceId()
+      await runCommand(binChectl, ['workspace:start', workspaceId, `-n ${NAMESPACE}`, '--telemetry=off'])
+      expect(await helper.waitWorkspaceStatus('RUNNING', WORKSPACE_START_TIMEOUT_MS)).to.equal(true)
+    })
+  })
+
+  describe('Test Che downgrade', () => {
+    it('Downgrade Che', async () => {
+      await runCommand(binChectl, ['server:update', '-y', `--version=${cheVersion}`, `-n ${NAMESPACE}`, '--telemetry=off'])
+      expect(await waitForCheServerImageTag(cheVersion, UPDATE_CHE_TIMEOUT_MS)).equal(true)
+    })
+
+    it('Check downgraded Che version', async () => {
+      expect(await waitForVersionInCheCR(cheVersion, CHE_VERSION_TIMEOUT_MS)).equal(true)
+    })
+  })
+
+})
+
+async function loginTest() {
```

Comment:
Should be moved to utility class.
May be `login`,  `runLoginCommand` or `loginIntoEclipseChe` ?